pennylane/decomposition/decomposition_rule.py

```diff
@@ -158,6 +163,7 @@ def register_resources(
             "Dynamic Allocation of Work Wires" section below.
         qfunc (Callable): the quantum function that implements the decomposition. If ``None``,
             returns a decorator for acting on a function.
+        heuristic (bool): whether the resources are estimated heuristically or computed exactly.
```

Comment:
Can we also make it clear that this is only relevant for testing and validation purposes?